src/java/org/tensorics/core/tensor/options/ExactShapesOrOneZeroStrategy.java

```diff
@@ -34,6 +34,10 @@
  */
 public class ExactShapesOrOneZeroStrategy implements ShapingStrategy {
 
+    public static ExactShapesOrOneZeroStrategy getInstance() {
```

Comment:
why do you need this? just to avoid the new operator, or is there any more specific reason?

In principle I fully agree that factory methods are a good idea, although here it conveys a very little information compared to a 'new XXX()'.